collectors/node_test.go

```diff
@@ -263,6 +265,29 @@ func TestNodeCollector(t *testing.T) {
 			`,
 			metrics: []string{"kube_node_status_condition"},
 		},
+		// Verify SpecTaints
+		{
+			nodes: []v1.Node{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "127.0.0.1",
+					},
+					Spec: v1.NodeSpec{
+						Taints: []v1.Taint{
+							{Key: "node.kubernetes.io/memory-pressure", Value: "true", Effect: v1.TaintEffectPreferNoSchedule},
+							{Key: "Accelerated", Value: "gpu", Effect: v1.TaintEffectPreferNoSchedule},
+							{Key: "Dedicated", Effect: v1.TaintEffectPreferNoSchedule},
+						},
+					},
+				},
+			},
+			want: metadata + `
+				kube_node_spec_taint{effect="PreferNoSchedule",key="Dedicated",node="127.0.0.1",value=""} 1
```

Comment:
@brancz Does Prometheus allows tag value to be an empty string?

Reply:
i thought this would work similar to labels whose value also could be the empty string.

https://github.com/kubernetes/kubernetes/blob/8d7d7a5e0d4d7e75f5a860574346944b8cc0fc43/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L88

it would be great to confirm as the taint value associated with the taint key is optional.

Reply:
Labels with empty values are thrown away at ingestion time, so yes it's perfectly fine to have that.